inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -220,9 +220,12 @@ public function get_sitemap_links( $type, $max_entries, $current_page ) {
 			return $links;
 		}
 
-		$stacked_urls     = array();
 		$posts_to_exclude = $this->get_excluded_posts();
 
+		if ( $post_type === 'page' && $this->get_page_on_front_id() ) {
```

Comment:
I think this can be moved to the `get_excluded_posts` method. To keep that responsibility together.